From 5bad302f1e508b00bfc03c3dbe30e3f8331677c9 Mon Sep 17 00:00:00 2001
From: Sandor Yu <Sandor.yu@nxp.com>
Date: Tue, 11 Sep 2018 16:11:12 +0800
Subject: [PATCH 4580/5242] MLK-19542: HDP: Fix iMX8QM 4Kp60 failed work with
 HDR10 TV

commit  4400a7ab8b880ce40618822d5867ca81d47fdc78 from
https://source.codeaurora.org/external/imx/linux-imx.git

HDP diver force output 10bit YUV420 for HDR10 TV.
But iMX8QM didn't support such function.
So add SOC version check only enable the function for iMX8MQ.

Signed-off-by: Sandor Yu <Sandor.yu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/hdp/imx-hdp.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/imx/hdp/imx-hdp.c b/drivers/gpu/drm/imx/hdp/imx-hdp.c
index 66ab798..bf48eb8 100644
--- a/drivers/gpu/drm/imx/hdp/imx-hdp.c
+++ b/drivers/gpu/drm/imx/hdp/imx-hdp.c
@@ -680,7 +680,8 @@ bool imx_hdp_bridge_mode_fixup(struct drm_bridge *bridge,
 	if (vic < 0)
 		return false;
 
-	if (vic == VIC_MODE_97_60Hz &&
+	/* force output 10bit YUV420 if HDMI sink support HDR10 */
+	if (vic == VIC_MODE_97_60Hz && cpu_is_imx8mq() &&
 	    (di->color_formats & DRM_COLOR_FORMAT_YCRCB420) &&
 	    (di->hdmi.y420_dc_modes & DRM_EDID_YCBCR420_DC_30)) {
 		hdp->bpc = 10;
-- 
1.7.9.5

